### DISCUSSION TITLE
Avalonia.Browser cannot run correctly

### DISCUSSION THREAD

**Discussion by 292390450** [AvaloniaUI/Avalonia#16472] [Q&A] [answered]
### Describe the bug

I used the command `dotnet new avalonia.xplat`to create a project that included Browser , Run Browser  Project, The webpage cannot be opened normally
![image](https://github.com/user-attachments/assets/b5b4ef8d-29fd-4759-8ed9-8f56603520f7)

### To Reproduce

1. dotnet new avalonia.xplat
2. cd the  Browser project
3. dotnet run 
![image](https://github.com/user-attachments/assets/6d6c98bc-cac8-43f5-96f1-56b503d58b9d)
4.Hold down Ctrl and click   `http://localhost:5235/`
5.
![image](https://github.com/user-attachments/assets/b5b4ef8d-29fd-4759-8ed9-8f56603520f7)

### Expected behavior

I am executing ` dotnet serve -p 5000` command in the published folder
![image](https://github.com/user-attachments/assets/3407e56b-ddb2-411c-98de-33c1536fb5aa)
Open address `http://localhost:5000`
Still lacking many files
![image](https://github.com/user-attachments/assets/29b5fa81-a4fd-4208-9616-11b89db166db)

### Avalonia version

11.1.0

### OS

Windows

**Comment by maxkatz6** [comment]
Make sure you have latest .NET SDK and workloads installed.

**Reply by maxkatz6** [reply]
```
PS E:\Work\Projects> mkdir browsertest
PS E:\Work\Projects> cd .\browsertest\
PS E:\Work\Projects\browsertest> dotnet new avalonia.xplat
The template "Avalonia Cross Platform Application" was created successfully.
PS E:\Work\Projects\browsertest> cd .\browsertest.Browser\
PS E:\Work\Projects\browsertest\browsertest.Browser> dotnet run
WasmAppHost --use-staticwebassets --runtime-config E:\Work\Projects\browsertest\browsertest.Browser\bin\Debug\net8.0-browser\browsertest.Browser.runtimeconfig.json

App url: http://localhost:5235/
App url: https://localhost:7169/
Debug at url: http://localhost:5235/_framework/debug
Debug at url: https://localhost:7169/_framework/debug
```

![image](https://github.com/user-attachments/assets/5034980c-d977-43e8-a90f-e6f24b0d089f)

**Reply by maxkatz6** [reply]
And
```
PS E:\Work\Projects\browsertest\browsertest.Browser> dotnet publish
Restore complete (0.4s)
You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy
  browsertest succeeded (0.6s) → E:\Work\Projects\browsertest\browsertest\bin\Release\net8.0\browsertest.dll
  browsertest succeeded (0.1s) → E:\Work\Projects\browsertest\browsertest\bin\Release\net8.0\browsertest.dll
  browsertest.Browser succeeded with 5 warning(s) (18.6s) → bin\Release\net8.0-browser\publish\
Build succeeded with 5 warning(s) in 36.5s
PS E:\Work\Projects\browsertest\browsertest.Browser> cd .\bin\Release\net8.0-browser\publish\wwwroot\

    Directory: E:\Work\Projects\browsertest\browsertest.Browser\bin\Release\net8.0-browser\publish\wwwroot

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d-----         7/25/2024   6:26 PM                _framework
-a----         7/25/2024   6:21 PM           1069 app.css
-a----         7/25/2024   6:21 PM         176111 favicon.ico
-a----         7/25/2024   6:21 PM          11135 index.html
-a----         7/25/2024   6:21 PM            436 main.js
PS E:\Work\Projects\browsertest\browsertest.Browser\bin\Release\net8.0-browser\publish\wwwroot> dotnet serve -p
 5000
Starting server, serving .
Listening on:
  http://localhost:5000
```
![image](https://github.com/user-attachments/assets/bf7836b6-68b2-4228-80b9-60a4a5dd14aa)

**Reply by maxkatz6** [reply]
> I am executing  dotnet serve -p 5000 command in the published folder

"avalonia.js" file should be inside of "_framework" folder. You either have old templates version, or your old installation .NET 8.0.100 had some kind of a bug - please make sure that both are latest.

**Reply by 292390450** [reply]
I created the project using the latest template 11.1.0, and the .net  SDK is also the latest
![image](https://github.com/user-attachments/assets/d78c8f62-04f9-4874-9c8c-905cf8ede650)
😔😔

**Reply by maxkatz6** [reply]
Run `dotnet workload update`.

**Comment by psulek** [comment, marked as answer]
I can confirm, for me `dotnet workload update` helps for me.

**Comment by cbordeman** [comment]
This doesn't work for me.  I have the latest templates, latest .net 9 SDK, installed the wasm workload and and I've done dotnet workload update.  Then I regenerated the entire solution from scratch using the latest templates.  It's unchanged and yet that avalonia.js is not copied in.

WTH!!

**Comment by nref** [comment]
This happened to me on macOS Rider from the template after I had just installed .NET 9 and updated the Avalonia templates with `dotnet new install Avalonia.Templates` but had not updated the workloads. `sudo dotnet workload install wasm-tools` fixed it for me. Didn't have to generate a new project or even restart Rider. The next build used the new workload and the launched page worked.

**Comment by luoczeng** [comment]
Same thing for me. Even the official sample ControlCatalog.Browser only displays an empty page except a 'Powered by Avalonia' text.
Tried several different browsers, same result.

**Reply by luoczeng** [reply]
![image](https://github.com/user-attachments/assets/cdbc1c2d-47ae-483f-a70b-141d3961b1f6)

**Reply by luoczeng** [reply]
![image](https://github.com/user-attachments/assets/93724514-74ec-45a2-8aca-a4af39ca6a07)
![image](https://github.com/user-attachments/assets/1b934858-87da-4bd4-988c-263679b996ba)
